src/main/java/com/compilercharisma/chameleonbusinessstudio/service/AppointmentService.java

```diff
@@ -25,177 +27,192 @@
+//    }
+//
+//    public Page<AppointmentEntity> getAvailableAppointments(LocalDateTime startTime, LocalDateTime endTime, Pageable page){
+//        return repo.findAll(occursWithin(startTime, endTime).and(isAvailable()), page);
+//    }
+//
+//    public void updateAppointment(AppointmentEntity appt){
+//        // this might also send notifications to users subscribed to the appointment
+//        repo.save(appt);
+//    }
+//
+//    /**
+//     * Checks if the given appointment is valid, and thus can be stored.
+//     *
+//     * @param e the appointment to validate
+//     * @return whether or not the appointment is valid
+//     */
+//    public boolean isAppointmentValid(AppointmentEntity e){
+//        return validator.isValid(e);
+//    }
+//
+//    /**
+//     * Checks if the given appointment is valid, throwing an
+//     * InvalidAppointmentException if it isn't.
+//     *
+//     * @param e the appointment to validate
+//     */
+//    public void validateAppointment(AppointmentEntity e){
+//        validator.validate(e);
+//    }
+//
+//    public void registerUser(AppointmentEntity appt, String email){
+//        if(!isSlotAvailable(appt)){
+//            throw new UnsupportedOperationException(String.format("Cannot register \"%s\" in appointment#%d: no slots available", email, appt.getId()));
+//        }
+//        Set<String> oldReg = appt.getRegisteredUsers();
+//        oldReg.add(email);
+//        appt.setRegisteredUsers(oldReg);
+//        repo.save(appt);
+//    }
+//
+//    /**
+//     * Checks if the given appointment can accept more users.
+//     *
+//     * @param appt the appointment to check
+//     * @return whether or not the appointment has any slots available
+//     */
+//    public boolean isSlotAvailable(AppointmentEntity appt){
+//        return appt.getTotalSlots() > appt.getRegisteredUsers().size();
+//    }
+//
+//    /**
+//     * Adds the given tag to this appointment
+//     *
+//     * @param appt
+//     * @param tag the tag key
+//     */
+//    public void addTag(AppointmentEntity appt, String tag){
+//        Set<String> copyOfOldTags = appt.getTags();
+//        copyOfOldTags.add(tag);
+//        appt.setTags(copyOfOldTags);
+//    }
+//
+//    /**
+//     * Checks if the given appointment has the given tag attached to it.
+//     *
+//     * @param appt the appointment to check
+//     * @param tag the tag to look for
+//     * @return whether the given appointment has the given tag
+//     */
+//    public boolean hasTag(AppointmentEntity appt, String tag){
+//        return appt.getTags().contains(tag);
+//    }
+//
+//    public boolean isUserRegistered(AppointmentEntity appt, String email){
+//        return appt.getRegisteredUsers().contains(email);
+//    }
+//
+//    public Mono<Appointment> createAppointment(Appointment appointment)
+//    {
+//        return Mono.just(appointment).flatMap(u -> appointmentRepository.createAppointment(appointment));
+//    }
```

Comment:
As per above. Perhaps we should merge this branch into a `dev` branch until we've reimplemented the repository features.